Sources/Build/Sanitizers.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+import Basic
+import Utility
+
+/// A set of enabled runtime sanitizers.
+public struct EnabledSanitizers {
+    public enum Sanitizer: String {
+        case address
+        case thread
+    }
+
+    /// A set of enabled sanitizers.
+    private var sanitizers: Set<Sanitizer> = []
+
+    public init() {}
+    public init(from list: [Sanitizer]) {
+        self.sanitizers = Set(list)
+    }
+
+    /// Return an established short name for a sanitizer, e.g. "asan".
+    private func shortname(_ sanitizer: Sanitizer) -> String {
+        switch sanitizer {
+        case .address: return "asan"
+        case .thread: return "tsan"
+        }
+    }
+
+    public func runtimeEnvironment(toolchain: Toolchain) -> [String: String] {
+      #if os(macOS)
+        return macosRuntimeEnvironment(toolchain: toolchain)
+      #else
+        return Process.env
+      #endif
+    }
+
+    public func macosRuntimeEnvironment(toolchain: Toolchain) -> [String: String] {
+        var env = Process.env
+        var insertLibs: [String] = []
+
+        if let existingLibs = env["DYLD_INSERT_LIBRARIES"] {
+            insertLibs = [existingLibs]
+        }
+
+        for sanitizer in sanitizers {
+            let libPath = sanitizerLibrary(clangPath: toolchain.clangCompiler, for: sanitizer)
+            insertLibs.append(libPath.asString)
+        }
+
+        env["DYLD_INSERT_LIBRARIES"] = insertLibs.joined(separator: ":")
+
+        if Process.verbose {
+            env["DYLD_PRINT_LIBRARIES"] = "1"
+        }
+
+        return env
+    }
+
+    /// Sanitization flags for the Swift compiler
+    public func compileSwiftFlags() -> [String] {
+        return sanitizers.map { "-sanitize=\($0.rawValue)" }
```

Comment:
nit: The map doesn't have side-effects, prefer: `return sanitizers.map({ "-sanitize=\($0.rawValue)" })`